internal/oapispec/ffi2swagger.go

```diff
@@ -35,35 +38,101 @@ func NewFFISwaggerGen() FFISwaggerGen {
 	return &ffiSwaggerGen{}
 }
 
-func (og *ffiSwaggerGen) Generate(ctx context.Context, baseURL string, ffi *fftypes.FFI) (swagger *openapi3.T, err error) {
-
-	// routes := []*Route{}
-
-	// for _, method := range ffi.Methods {
-	// 	if routes, err = og.addMethod(ctx, routes, method); err != nil {
-	// 		return nil, err
-	// 	}
-	// }
-
-	// for _, event := range ffi.Events {
-	// 	if routes, err = og.addEvent(ctx, routes, event); err != nil {
-	// 		return nil, err
-	// 	}
-	// }
-
-	return nil, nil
+func (og *ffiSwaggerGen) Generate(ctx context.Context, baseURL string, api *fftypes.ContractAPI, ffi *fftypes.FFI) (swagger *openapi3.T, err error) {
+	hasLocation := api.Location.String() != "null"
+
+	routes := []*Route{}
+	for _, method := range ffi.Methods {
+		routes = og.addMethod(routes, method, hasLocation)
+	}
+	for _, event := range ffi.Events {
+		routes = og.addEvent(routes, event, hasLocation)
+	}
+
+	return SwaggerGen(ctx, routes, &SwaggerGenConfig{
+		Title:       ffi.Name,
+		Version:     ffi.Version,
+		Description: ffi.Description,
+		BaseURL:     baseURL,
+	}), nil
 }
 
-// func (og *ffiSwaggerGen) addMethod(ctx context.Context, routes []*Route, method *fftypes.FFIMethod) ([]*Route, error) {
+func (og *ffiSwaggerGen) addMethod(routes []*Route, method *fftypes.FFIMethod, hasLocation bool) []*Route {
+	return append(routes, &Route{
```

Comment:
I assume the `query` method will come in a future PR.

Reply:
Yes, that work is still pending. Since the API doesn't exist, no point in generating the spec for it.